beacon-chain/blockchain/forkchoice_update_execution.go

```diff
@@ -93,16 +79,10 @@ func (s *Service) forkchoiceUpdateWithExecution(ctx context.Context, newHeadRoot
 		return errors.Wrap(err, "could not notify forkchoice update")
 	}
 
-	if shouldUpdate {
-		if err := s.saveHead(ctx, newHeadRoot, headBlock, headState); err != nil {
-			log.WithError(err).Error("could not save head")
-		}
-
-		// Only need to prune attestations from pool if the head has changed.
-		if err := s.pruneAttsFromPool(headBlock); err != nil {
-			return err
-		}
+	if err := s.saveHead(ctx, newHeadRoot, headBlock, headState); err != nil {
+		log.WithError(err).Error("could not save head")
 	}
 
-	return nil
+	// Only need to prune attestations from pool if the head has changed.
+	return s.pruneAttsFromPool(headBlock)
```

Comment:
would it be possible to refactor and return the headblock,isNewHead,isNewProposer instead? this way we can also use it for the postBlockOperations?

Reply:
I dont think we should alter the signature. Some options are:
1. call `postBlockOperations` here
2. call `postBlockOperations` downstream but return nil on non-canonical block